src/crdt.rs

```diff
@@ -590,33 +590,45 @@ impl Crdt {
     }
     fn run_window_request(
         window: &Window,
+        me: &ReplicatedData,
         from: &ReplicatedData,
         ix: u64,
         blob_recycler: &BlobRecycler,
     ) -> Option<SharedBlob> {
         let pos = (ix as usize) % window.read().unwrap().len();
         if let Some(blob) = &window.read().unwrap()[pos] {
-            let rblob = blob.read().unwrap();
-            let blob_ix = rblob.get_index().expect("run_window_request get_index");
+            let mut wblob = blob.write().unwrap();
+            let blob_ix = wblob.get_index().expect("run_window_request get_index");
             if blob_ix == ix {
+                let num_retransmits = wblob.meta.num_retransmits;
+                wblob.meta.num_retransmits += 1;
+                
+                if me.current_leader_id == me.id &&
+                   num_retransmits != 0 && 
+                   !num_retransmits.is_power_of_two()
+                {
+                    return None;
```

Comment:
If it's not power of two, shouldn't we transmit with id of receiver (the original logic)?

Reply:
good catch, will update soon

Reply:
Cool, while at it, could you fix the code formatting error (causing CI failure)

Reply:
Hm, I misunderstood this part of the original issue. Even if the goal is to reduce network congestion among the validators, it still does make more sense to just respond with a response, but not rebroadcast, on non-power of two requests. Good point!

Reply:
Sure, done and done :)